app/uk/gov/hmrc/exports/services/reversemapping/declaration/locations/OriginationCountryParser.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.services.reversemapping.declaration.locations
+
+import uk.gov.hmrc.exports.models.StringOption
+import uk.gov.hmrc.exports.models.declaration.Country
+import uk.gov.hmrc.exports.services.reversemapping.MappingContext
+import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser
+import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser.XmlParserResult
+import uk.gov.hmrc.exports.services.reversemapping.declaration.XmlTags._
+
+import javax.inject.Singleton
+import scala.xml.NodeSeq
+
+@Singleton
+class OriginationCountryParser extends DeclarationXmlParser[Option[Country]] {
+
+  override def parse(inputXml: NodeSeq)(implicit context: MappingContext): XmlParserResult[Option[Country]] = Right(
+    StringOption((inputXml \ Declaration \ GoodsShipment \ ExportCountry \ ID).text)
+      .map(exportCountryId => Country(Some(exportCountryId)))
+  )
+}
```

Comment:
Nit: maybe you could combine this and the DestinationCountryParser into a single common CountryParser class that is passed in a filtered NodeSeq by the calling LocationsParser (similar to how PartiesParser calls the common EntityDetailsParser class multiple times for different values)